sources/s3/s3.go

```diff
@@ -363,12 +370,19 @@ func getImpl(ctx context.Context, client S3Client, scope string, query string) (
 		if lambdaConfig.LambdaFunctionArn != nil {
 			if a, err = sources.ParseARN(*lambdaConfig.LambdaFunctionArn); err == nil {
 				// +overmind:link lambda-function
-				item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{Query: &sdp.Query{
-					Type:   "lambda-function",
-					Method: sdp.QueryMethod_SEARCH,
-					Query:  *lambdaConfig.LambdaFunctionArn,
-					Scope:  sources.FormatScope(a.AccountID, a.Region),
-				}})
+				item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+					Query: &sdp.Query{
+						Type:   "lambda-function",
+						Method: sdp.QueryMethod_SEARCH,
+						Query:  *lambdaConfig.LambdaFunctionArn,
+						Scope:  sources.FormatScope(a.AccountID, a.Region),
+					},
+					BlastPropagation: &sdp.BlastPropagation{
+						// Tightly coupled
+						In:  true,
+						Out: true,
```

Comment:
In this case the function is invoked when things in the bucket change (like a file being added). If we were to change the config of the bucket potentially this could break the processing.

Reply:
hmmmm .. ok